test/integration/nodeagent/lease/lease_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package lease_test
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/utils/clock/testing"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+
+	leasecontroller "github.com/gardener/gardener/pkg/nodeagent/controller/lease"
+	gardenerutils "github.com/gardener/gardener/pkg/utils"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("Reconcile", func() {
+	Describe("Lease controller tests", func() {
+		var (
+			node      *corev1.Node
+			nodeName  string
+			fakeClock *testing.FakeClock
+		)
+
+		BeforeEach(func() {
+			nodeName = "test-" + gardenerutils.ComputeSHA256Hex([]byte(uuid.NewUUID()))[:8]
+			fakeClock = testing.NewFakeClock(time.Now().UTC())
+
+			By("Setup manager")
+			mgr, err := manager.New(restConfig, manager.Options{
+				Metrics: metricsserver.Options{BindAddress: "0"},
+				Cache: cache.Options{
+					DefaultNamespaces: map[string]cache.Config{testNamespace.Name: {}},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Register controller")
+			leaseReconciler := &leasecontroller.Reconciler{
+				Clock:                fakeClock,
+				NodeName:             nodeName,
+				Namespace:            testNamespace.Name,
+				RenewIntervalSeconds: 3,
+			}
+			Expect(leaseReconciler.AddToManager(mgr)).To(Succeed())
+
+			By("Start manager")
+			mgrContext, mgrCancel := context.WithCancel(ctx)
+
+			go func() {
+				defer GinkgoRecover()
+				Expect(mgr.Start(mgrContext)).To(Succeed())
+			}()
+
+			DeferCleanup(func() {
+				By("Stop manager")
+				mgrCancel()
+			})
```

Comment:
We could move this part to `lease_suite_test.go` since it's enough to do this once per test suite.